### PR TITLE
Consolidate browser compatibility data, drop IE11 support

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary
* Consolidates browser compatibility data so that we use the same source of truth for browserslist and checking supported browsers in the frontend
* Drops IE11 support
* Updates deprecation warnings about IE11 for browsing users to give upgrade warning instead

## References
Fixes https://github.com/learningequality/kolibri/issues/5336

## Reviewer guidance
The only other place I could find IE11 specific code was in the TextTruncatorCSS component, but I was not entirely sure how to remove the IE11 specific code. Advice from @MisRob would be appreciated, or we can file a follow up issue for this cleanup.

----

## Testing checklist

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/core/assets/src/utils/minimumBrowserRequirements.js [hunk 1]
```diff
@@ -1,16 +1,30 @@
...
-  },
-};
+const minimumBrowserRequirements = {};
+
+const browserRegex = /^([a-zA-Z]+) ([><=]+) (\d+)(?:\.(\d+))?(?:\.(\d+))?$/;
+
+for (const browser of browsers) {
+  const [name, sign, major, minor, patch] = browserRegex.exec(browser).slice(1);
+  // This only supports > and >=, but that's all we need.
+  const addOne = sign === '>' ? 1 : 0;
```
Comment (marcellamaki) [inline kolibri/core/assets/src/utils/minimumBrowserRequirements.js]: `sign` here means literally `>` or `>=`?

**Review by marcellamaki — CHANGES_REQUESTED** [review]
The changes seem okay, but the browser requirements code is hard to follow. Added a comment about what in particular tripped me up, but happy to chat further about what might help

File: kolibri/core/assets/src/utils/minimumBrowserRequirements.js [hunk 2]
```diff
@@ -1,16 +1,30 @@
...
+  const entry = {
+    major: isUndefined(minor) && isUndefined(patch) ? Number(major) + addOne : Number(major),
```
Comment (marcellamaki) [inline kolibri/core/assets/src/utils/minimumBrowserRequirements.js]: My other comment as part of the review seems to have disappeared, so returning to this to add it in. 

The combination of the ternary operator, and the fact that `addOne` can be `1` or `0` (and isn't actually always 1) makes this hard for me to reason about. I think either adding a comment with a practical example, or adjusting the name of `addOne` (I have been trying to come up with a specific suggestion and have not so far, but will continue to ponder) might help make this clearer

File: kolibri/core/assets/src/utils/minimumBrowserRequirements.js (hunk 1 above)
Comment (rtibbles) [reply kolibri/core/assets/src/utils/minimumBrowserRequirements.js]: Yes, I hated this, and only resorted to this to avoid about 15 nested if statements. Will think about how to rework for clarity.

**Comment by rtibbles** [comment]
Have updated to try to make the parsing logic more self explanatory.

**Review by marcellamaki — APPROVED** [review]
Thanks for breaking this down - I find it much easier to comprehend. Per slack discussion, we'll go forward with this and check in with @MisRob re: TextTruncator and address in a follow up issue as appropriate

**Comment by rtibbles** [comment]
Follow up will happen in KDS where KTextTruncator has been implemented.

**Comment by rtibbles** [comment]
Docs build failure is unrelated, merging.
